octoprint_mrbeam/static/js/material_settings.js

```diff
@@ -347,37 +396,82 @@ $(function () {
+                type: null,
+                url: null
             },
             "Wood Sticker": {
                 name: gettext("Wood Sticker"),
                 img: "Wood-Sticker.jpg",
                 description: "",
                 hints: "",
                 safety_notes: "",
+                type: "product",
+                url: gettext("https://www.mr-beam.org/en/products/edelholzsticker")
             },
             "Solid Wood": {
                 name: gettext("Solid Wood"),
                 img: "Solid-Wood.jpg",
                 description: "",
                 hints: "",
                 safety_notes: "",
+                type: "collection",
+                url: gettext("https://www.mr-beam.org/en/collections/holz")
             },
             "Stainless Steel": {
                 name: gettext("Stainless Steel"),
                 img: "Stainless-Steel.jpg",
                 description: "",
                 hints: gettext("Engravings on stainless steel only work with our \"mark solid\" spray. Use line distance 0,1mm for the best result."),
                 safety_notes: gettext("Do not engrave stainless steel without the application of laser marking color."),
+                type: null,
+                url: null
             },
         };
         ///// EDIT MATERIAL SETTINGS ABOVE THIS LINE ////////`
+
+        self.constructShopifyURL = function (materialName, materialColor) {
+            // Check if material exists in the Database
+            if(materialName in self.materialSettingsDatabase) {
+                // Check if a URL exists for this material
+                if("url" in self.materialSettingsDatabase[materialName] &&
+                self.materialSettingsDatabase[materialName]["url"] &&
+                typeof self.materialSettingsDatabase[materialName]["url"] === "string"){
+                    return self.addUrlReferral(self.constructMaterialURL(materialName, materialColor))
+                }
+            }
+            return self.addUrlReferral(self.MATERIAL_SHOPIFY_LINK);
+        }
+
+        self.constructMaterialURL = function (materialName, materialColor) {
+            const url = self.materialSettingsDatabase[materialName]["url"];
+            // Check if a color variant query parameter exists for this URL
+            if("variant" in self.materialSettingsDatabase[materialName] &&
+            "color" in self.materialSettingsDatabase[materialName]["variant"] &&
+            materialColor in self.materialSettingsDatabase[materialName]["variant"]["color"]){
+                const variantParameterKey = "variant"
+                const variantParameterValue = self.materialSettingsDatabase[materialName]["variant"]["color"][materialColor];
+                return url + "?" + variantParameterKey + "=" + variantParameterValue;
+            } else {
+                return url;
+            }
+        }
+
+        self.addUrlReferral = function (url){
+            if(url.includes("?")){
```

Comment:
might be worth it making this a ternary... like:
`param_separator = url.includes("?") ? "&" : "?" ;`
`return url + param_separator + "utm_source=mrbeam_device_frontend"`

Reply:
It definitely is (Y) 
done